integrated_channels/cornerstone/views.py

```diff
@@ -209,20 +124,32 @@ def get(self, request, *args, **kwargs):
         )
         exporter = enterprise_config.get_content_metadata_exporter(worker_user)
         transmitter = enterprise_config.get_content_metadata_transmitter()
-        data = transmitter.transmit(*exporter.export_force_all_catalogs())
+
+        get_count = int(request.GET.get('count', exporter.MAX_PAYLOAD_COUNT))
+        content_item_count = min(get_count, exporter.MAX_PAYLOAD_COUNT)
+        data = transmitter.transmit(*exporter.export_for_web_polling(max_payload_count=content_item_count))
 
         logger.info(f'integrated_channel=CSOD, '
                     f'integrated_channel_enterprise_customer_uuid={enterprise_customer_uuid}, '
                     f'transmitting {len(data)} items.')
 
-        if len(data) > 0:
-            minLastModifiedContent = min(data, key=lambda x: x['LastModifiedUTC'])
-            maxLastModifiedContent = max(data, key=lambda x: x['LastModifiedUTC'])
-            # If-Modified-Since compared to content logging
-            logger.info(f'integrated_channel=CSOD, '
-                        f'integrated_channel_enterprise_customer_uuid={enterprise_customer_uuid}, '
-                        f'If-Modified-Since={request.headers.get("If-Modified-Since")}, '
-                        f'minLastModifiedUTC={minLastModifiedContent.get("LastModifiedUTC")}, '
-                        f'maxLastModifiedUTC={maxLastModifiedContent.get("LastModifiedUTC")}')
+        if_modified_since = request.headers.get("If-Modified-Since")
+        if_modified_since = if_modified_since and parse_http_date_safe(if_modified_since)
+
+        # our CSOD imp. only supports creates/updates, no deletes
+        # fib and make it appear like all creates/updates are past if-modified-since
+        if if_modified_since:
+            for item in data:
+                this_last_modified = parser.parse(item['LastModifiedUTC'])
+                if if_modified_since > int(this_last_modified.timestamp()):
+                    # log when we find an incorrect modified date, then "fix" it
+                    logger.info(f'integrated_channel=CSOD, '
+                                f'integrated_channel_enterprise_customer_uuid={enterprise_customer_uuid}, '
+                                f'If-Modified-Since={request.headers.get("If-Modified-Since")}, '
+                                f'integrated_channel_course_key={item["ID"]}, '
+                                f'LastModifiedUTC={item["LastModifiedUTC"]} '
+                                f'fixing modified/header mismatch')
+                    if_modified_since_dt = datetime.datetime.fromtimestamp(if_modified_since)
```

Comment:
we track when we need to update content with the audit records, the if-modified-since can be faked a bit until we have full support with the soft-deletes.